src/cli_util.rs

```diff
@@ -1848,6 +1848,28 @@ impl CliRunner {
         }
     }
 
+    /// Registers new global arguments in addition to the default ones.
+    pub fn add_global_args<A, F>(self, process_before: F) -> Self
+    where
+        A: clap::Args,
+        F: FnOnce(&mut Ui, A) -> Result<(), CommandError> + 'static,
```

Comment:
Please let me know if we need to remove `'static`. It should be doable, but I didn't try for subcommands because it's unlikely a command dispacher function captures stack variables.

Reply:
I don't think we'll need it for our case. In our case we call a function that sets global state for the C++ flag parser, so we shouldn't need any data other than the custom global args. Thanks for asking.